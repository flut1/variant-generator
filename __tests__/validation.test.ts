import createVariants from '../src';

describe('createVariants()', () => {
  test('throws on yields with different lengths', () => {
    expect(() => {
      const inputs = [[1, 2, 3], [1, 2, 3, 4]];
      let pickInputAtIndex = 0;

      function getInput() {
        pickInputAtIndex = pickInputAtIndex ? 0 : 1;
        return inputs[pickInputAtIndex];
      }

      const resultIterator = createVariants(function*() {
        return [yield getInput()];
      });

      // iterate limited number of times to prevent infinite loop
      resultIterator.next();
      resultIterator.next();
      resultIterator.next();
      resultIterator.next();
      resultIterator.next();
    }).toThrowErrorMatchingSnapshot();
  });
  test('throws on yields with different types', () => {
    expect(() => {
      const inputs = [[1, 2, 3], [1, 2, 'foo']];
      let pickInputAtIndex = 0;

      function getInput() {
        pickInputAtIndex = pickInputAtIndex ? 0 : 1;
        return inputs[pickInputAtIndex];
      }

      const resultIterator = createVariants(function*() {
        return [yield getInput()];
      });

      // iterate limited number of times to prevent infinite loop
      resultIterator.next();
      resultIterator.next();
      resultIterator.next();
      resultIterator.next();
      resultIterator.next();
    }).toThrowErrorMatchingSnapshot();
  });
  test('throws on a yield with a non-array', () => {
    expect(() => {
      const test = Array.from(
        createVariants(function*() {
          return yield { 0: 'foo', 1: 'bar' };
        }),
      );
    }).toThrowErrorMatchingSnapshot();
  });
});
