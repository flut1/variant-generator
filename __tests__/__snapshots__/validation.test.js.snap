// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createVariants() throws on a yield with a non-array 1`] = `"yielded values in createVariants() should be Arrays of options. Got object"`;

exports[`createVariants() throws on yields with different lengths 1`] = `
"The generator passed to createVariants() seems to yield non-deterministic input Arrays.
The generator should yield the same inputs on each call. Yield #0 was inconsistent:
inputs previously had length 4, now yields 3"
`;

exports[`createVariants() throws on yields with different types 1`] = `
"The generator passed to createVariants() seems to yield non-deterministic input Arrays.
The generator should yield the same inputs on each call. Yield #0 was inconsistent:
input option 2 was of type string, is now number"
`;

exports[`createVariants() throws when passing a non-function 1`] = `"Unexpected argument passed to createVariants(). Expected a generator function, got object"`;

exports[`createVariants() throws when passing a non-generator function 1`] = `"The function passed to createVariants() does not seem to be a generator function. Please use function*"`;
